src/supernova/mod.rs

```diff
@@ -60,208 +116,205 @@ where
-  G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
-{
-  /// Indexed `RunningClaim`s
-  claims: Vec<RunningClaim<G1, G2, C1, C2>>,
-  /// Digest constructed from these `RunningClaim`s' parameters
-  // Once we serialize this struct, it's important to add this
-  // annotaiton to the digest field for it to be treated properly: #[serde(skip, default = "OnceCell::new")]
-  digest: OnceCell<G1::Scalar>,
-}
+    let PublicParams {
+      circuit_params,
+      ro_consts_primary,
+      ro_consts_circuit_primary,
+      ck_primary,
+      augmented_circuit_params_primary,
+      F_arity_secondary,
+      ro_consts_secondary,
+      ro_consts_circuit_secondary,
+      ck_secondary,
+      r1cs_shape_secondary,
+      augmented_circuit_params_secondary,
+      digest: _digest,
+      _p,
+    } = self;
 
-impl<G1, G2, C1, C2> Index<usize> for RunningClaims<G1, G2, C1, C2>
-where
-  G1: Group<Base = <G2 as Group>::Scalar>,
-  G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
-{
-  type Output = RunningClaim<G1, G2, C1, C2>;
+    let aux_params = AuxParams {
+      ro_consts_primary,
+      ro_consts_circuit_primary,
+      ck_primary,
+      augmented_circuit_params_primary,
+      F_arity_secondary,
+      ro_consts_secondary,
+      ro_consts_circuit_secondary,
+      ck_secondary,
+      r1cs_shape_secondary,
+      augmented_circuit_params_secondary,
+      digest,
+    };
 
-  fn index(&self, index: usize) -> &Self::Output {
-    &self.claims[index]
+    (circuit_params, aux_params)
   }
-}
 
-impl<G1, G2, C1, C2> Digestible for RunningClaims<G1, G2, C1, C2>
-where
-  G1: Group<Base = <G2 as Group>::Scalar>,
-  G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
-{
-  fn write_bytes<W: Sized + io::Write>(&self, byte_sink: &mut W) -> Result<(), io::Error> {
-    assert!(!self.claims.is_empty());
-    for claim in &self.claims {
-      claim.get_public_params().write_bytes(byte_sink)?;
+  /// Create a [PublicParams] from a vector of raw [CircuitParams] and auxillary params.
+  pub fn from_parts(circuit_params: Vec<CircuitParams<G1>>, aux_params: AuxParams<G1, G2>) -> Self {
+    PublicParams {
+      circuit_params,
+      ro_consts_primary: aux_params.ro_consts_primary,
+      ro_consts_circuit_primary: aux_params.ro_consts_circuit_primary,
+      ck_primary: aux_params.ck_primary,
+      augmented_circuit_params_primary: aux_params.augmented_circuit_params_primary,
+      F_arity_secondary: aux_params.F_arity_secondary,
+      ro_consts_secondary: aux_params.ro_consts_secondary,
+      ro_consts_circuit_secondary: aux_params.ro_consts_circuit_secondary,
+      ck_secondary: aux_params.ck_secondary,
+      r1cs_shape_secondary: aux_params.r1cs_shape_secondary,
+      augmented_circuit_params_secondary: aux_params.augmented_circuit_params_secondary,
+      digest: aux_params.digest.into(),
```

Comment:
The preferred way to do this would be to not trust the `AuxParams` digest and simply re-compute the digest, and compare it to the digest of the `AuxParams`. 
i.e. `assert_eq!(self.digest() == aux_params.digest)` which guarantees the correctness of the digest and that the user hasn't matched up the wrong set of circuits.
What you've implemented should pehaps be called `from_parts_unchecked` instead?